src/test/java/io/jenkins/plugins/analysis/warnings/recorder/pageobj/SourceCodeView.java

```diff
@@ -0,0 +1,133 @@
+    /** The xPath to the message describing the issue (starting at the main panel). */
+    public static final String ISSUE_MESSAGE_XPATH = "//pre//div//div";
+
+    /** The source code view html page. */
+    private HtmlPage sourceCodeViewPage;
+
+    /**
+     * Constructor.
+     *
+     * @param sourceCodeViewPage
+     *         The source code view html page.
+     */
+    public SourceCodeView(HtmlPage sourceCodeViewPage) {
+        this.sourceCodeViewPage = sourceCodeViewPage;
+    }
+
+    /**
+     * Extracts the name of the file that is displayed on the html page.
+     *
+     * @return The file name.
+     */
+    public String getFileName() {
+        String name;
+
+        List<DomNode> headingByXPath = getMainPanel().getByXPath(FILE_NAME_XPATH);
+        if (!headingByXPath.isEmpty()) {
+            String heading = headingByXPath.get(0).asText();
+            Matcher matcher = FILE_NAME_PATTERN.matcher(heading);
+            if (matcher.matches()) {
+                return matcher.group(1);
+            }
+        }
+
+        throw new NoSuchElementException("Can't find the file name on page.");
+    }
+
+    /**
+     * Extracts the whole source code displayed on the html page.
+     *
+     * @return The source code.
+     */
+    public String getSourceCode() {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<DomNode> sourceCodeList = getMainPanel().getByXPath(SOURCE_CODE_XPATH);
+
+        for (DomNode node : sourceCodeList) {
+            String code = node.asText();
+            stringBuilder.append(code);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Extracts the source code line that produces the issue.
+     *
+     * @return The source code line producing the issue.
+     */
+    public String getIssueSourceCodeLine() {
+        StringBuilder sourceCodeLineBuilder = new StringBuilder();
+        List<DomNode> issueSourceCodeList = getMainPanel().getByXPath(ISSUE_SOURCE_CODE_XPATH);
+
+        if (!issueSourceCodeList.isEmpty()) {
+            issueSourceCodeList.forEach(c -> sourceCodeLineBuilder.append(c.asText()));
+        }
+        else {
+            throw new NoSuchElementException("Can't find the issue source code line on page.");
+        }
+
+        return sourceCodeLineBuilder.toString();
+    }
+
+    /**
+     * Extracts the message about the issue from the html page.
+     *
+     * @return The issue message.
+     */
+    public String getIssueMessage() {
+        StringBuilder issueMessageBuilder = new StringBuilder();
+        List<DomNode> issueMessageList = getMainPanel().getByXPath(ISSUE_MESSAGE_XPATH);
+
+        if (!issueMessageList.isEmpty()) {
+            for (DomNode node : issueMessageList) {
+                String issueMessage = node.asText();
+
+                //replacing "new line" and "tab" character at the start of the message
+                issueMessage = issueMessage.replace("\t", "").replace("\n", "");
```

Comment:
would `StringUtils.trim()`also work?

Reply:
I couldn't reproduce the issue I had with the "new line" and "tab" character so I simplified the code and removed the string operations.